### DISCUSSION TITLE
Unexpectedly high free tier usage for Queues

### DISCUSSION THREAD

**Discussion by harveydobson** [brefphp/bref#1760] [Support] [answered]
Hi All, 

I've been playing around with Bref quite a lot since the PHP UK conference (thanks for a fantastic talk @mnapoli !!)

I've normally deployed my functions with a queue as for many of them I would like to, at some point, make use of the queue, and I had assumed as queues are charged by request, that to simply have the queue but not utilise it would be fine.

I don't believe I've had more than one function running at a time, apart from today...  Anyway earlier I got this email from AWS. 

![image](https://github.com/brefphp/bref/assets/8660605/f4dd7a95-ea9b-4c90-a134-0a9b0efa3c44)

I see that in the config I am using the `timeout: 60 # seconds` option from the example, so my thinking is even if this ran on a schedule, it would still be once a minute, which is just over 6000 invocations a day and not far off 180,000 a month.

I'm not getting any traffic to my functions, possibly some internet noise, and a handful of requests.

Would someone be able to explain a bit more about how these queue requests tally up?

Thanks,

Harvey.

**Comment by mnapoli** [comment, marked as answer]
Hi Harvey! This is how the SQS -> Lambda integration works. Under the hood, AWS Lambda maintains a "poller" that waits for SQS messages all the time. When it gets messages, it invokes our function.

That's how, for us the users, we don't have to poll SQS. AWS does it for us basically. The downside is that that creates continuous requests to SQS, and we are "charged" for those (the free tier kicks in though).

That could benefit from a mention in the Bref docs though!

**Reply by t-richard** [reply]
To reduce the effect, you can increase the `MaximumBatchingWindowInSeconds` but this can lead to delay in delivering your messages to the Lambda function when you have a batch size greater than `1` 

It basically tells the poller to wait for the given time to collect the largest possible batch it can. And this is counted as a single request to the SQS API.

**Reply by harveydobson** [reply]
Thanks both. Yes as you say, the free tier does account for most, if not all of the usage, and for anything over that, it's still a very minimal cost. My interest here was purely in understanding what is happening under the hood, thanks very much for clarifying that.
